apps/s_client.c

```diff
@@ -3138,6 +3149,8 @@ int s_client_main(int argc, char **argv)
     OPENSSL_free(bindstr);
     OPENSSL_free(host);
     OPENSSL_free(port);
+    if (thost != NULL) OPENSSL_free(thost);
```

Comment:
Please drop the ifs, the free call checks it.

Reply:
Done.